app/src/androidTest/java/com/github/sdpteam15/polyevents/AdminHubFragmentTest.kt

```diff
@@ -8,53 +11,113 @@ import androidx.test.espresso.intent.matcher.IntentMatchers
+import org.mockito.Mockito.`when` as When
 
 @RunWith(AndroidJUnit4::class)
 class AdminHubFragmentTest {
-    @Rule
-    @JvmField
     var mainActivity = ActivityScenarioRule(MainActivity::class.java)
+    lateinit var scenario : ActivityScenario<MainActivity>
+
+
+    lateinit var testUser: UserEntity
+
+    val uid = "testUid"
+    val username = "JohnDoe"
+    val email = "John@Doe.com"
 
     @Before
     fun setup() {
+        val mockedDatabase = mock(DatabaseInterface::class.java)
+        val mockedUserProfile = UserProfile("TestID","TestName")
+        When(mockedDatabase.currentProfile).thenReturn(mockedUserProfile)
+        When(mockedDatabase.getUpcomingEvents()).thenReturn(initEvents())
+        Database.currentDatabase = mockedDatabase
+
         FirebaseAuth.getInstance().signOut()
-        val user = mock(DatabaseUserInterface::class.java)
-        MainActivity.currentUser = User.Companion.invoke(user)
+        testUser = UserEntity(
+            uid = uid,
+            username = username,
+            email = email
+        )
+        MainActivity.currentUser = testUser
+
+        val intent = Intent(ApplicationProvider.getApplicationContext(), MainActivity::class.java)
+        scenario = ActivityScenario.launch(intent)
 
         Espresso.onView(ViewMatchers.withId(R.id.ic_home)).perform(click())
         Espresso.onView(ViewMatchers.withId(R.id.id_fragment_admin_hub))
             .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
-
         Intents.init()
     }
 
+    private fun initEvents(): MutableList<Event> {
+        val events: MutableList<Event> = mutableListOf()
+        events.add(
+            Event(
+                eventId = "event1",
+                eventName = "Sushi demo",
+                description = "Super hungry activity !",
+                startTime = LocalDateTime.of(2021, 3, 7, 12, 15),
+                organizer = "The fish band",
+                zoneName = "Kitchen",
+                tags = mutableSetOf("sushi", "japan", "cooking")
+            )
+        )
+        events.add(
+            Event(
+                eventId = "event2",
+                eventName = "Saxophone demo",
+                description = "Super noisy activity !",
+                startTime = LocalDateTime.of(2021, 3, 7, 17, 15),
+                organizer = "The music band",
+                zoneName = "Concert Hall"
+            )
+        )
+        events.add(
+            Event(
+                eventId = "event3",
+                eventName = "Aqua Poney",
+                description = "Super cool activity !" +
+                        " With a super long description that essentially describes and explains" +
+                        " the content of the activity we are speaking of.",
+                startTime = LocalDateTime.of(2021, 3, 7, 14, 15),
+                organizer = "The Aqua Poney team",
+                zoneName = "Swimming pool"
+            )
+        )
+        return events
+    }
```

Comment:
Why should we initialise the events since, if I'm correct, I didn't test anything related to the events in the admin hub tests

Reply:
@Rygone I refer to Robin for this

Reply:
currentDatabase.getUpcomingEvents() was called by the homepage before, but now its useless